File: 424-longest-repeating-character-replacement/424-longest-repeating-character-replacement.cpp
class Solution {
public:
    int characterReplacement(string s, int k) {
        int res = 0;
        int left = 0;
        int right = 0;
        vector<int> chars(26);
        int mostFregChar = 0;
        while (right < s.size()) {
            chars[s[right]-'A']++;
            mostFregChar = max(mostFregChar, chars[s[right]-'A']);
            if (right-left+1-mostFregChar <= k) {
                res = max(res, right-left+1);
            } else {
                chars[s[left++]-'A']--;
            }
            right++;
        }
        return res;
    }
};

File: 424-longest-repeating-character-replacement/424-longest-repeating-character-replacement.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        left = 0;
        right = 0;
        freqChar = 0;
        chars = {}
        res = 0
        while (right < len(s)):
            chars[s[right]] = 1 + chars.get(s[right], 0)
            freqChar = max(freqChar, chars[s[right]])
            
            if right-left+1-freqChar <= k:
                res = max(res, right-left+1)
            else:
                chars[s[left]] -= 1
                left += 1
            right += 1
        
        return res
            